### ISSUE TITLE
[Upgrade] [CLI] Failed on RoutesModule

### ISSUE THREAD

**Issue by Kaik** [zikula/core#2355]
Hi, I have tried upgrade and this error occured
```
[Kaik@kaikbig src]$ php app/console zikula:upgrade -vvv                                                                                                            
---------------------------
| Zikula Upgrader Script |
---------------------------
*** UPGRADING TO ZIKULA CORE v1.4.0 ***
Please enter the username of the new Zikula admin user default:[admin]: Kaik
Please enter the password of the new Zikula admin user: 
Please enter the host where you install Zikula, e.g. "example.com". Do not include subdirectories: ml.dev.com
Please enter the scheme of where you install Zikula, can be either "http" or "https" default:[http]: 
Please enter the url path of the directory where you install Zikula, leave empty if you install it at the top level. Example: /my/sub-dir default:[]: 
Login
Logged in as admin
Upgrade modules
Modules upgraded
Install Zikula Routes Module


                                                             
  [Symfony\Component\Debug\Exception\ContextErrorException]  
  Notice: Undefined offset: 0                                
                                                             


Exception trace:
 () at /home/Kaik/public_html/mojeleicester/src/lib/Zikula/Bundle/CoreInstallerBundle/Controller/AjaxUpgradeController.php:115
 Symfony\Component\Debug\ErrorHandler->handleError() at /home/Kaik/public_html/mojeleicester/src/lib/Zikula/Bundle/CoreInstallerBundle/Controller/AjaxUpgradeController.php:115
 Zikula\Bundle\CoreInstallerBundle\Controller\AjaxUpgradeController->installRoutesModule() at /home/Kaik/public_html/mojeleicester/src/lib/Zikula/Bundle/CoreInstallerBundle/Controller/AjaxUpgradeController.php:70
 Zikula\Bundle\CoreInstallerBundle\Controller\AjaxUpgradeController->executeStage() at /home/Kaik/public_html/mojeleicester/src/lib/Zikula/Bundle/CoreInstallerBundle/Controller/AjaxUpgradeController.php:54
 Zikula\Bundle\CoreInstallerBundle\Controller\AjaxUpgradeController->commandLineAction() at /home/Kaik/public_html/mojeleicester/src/lib/Zikula/Bundle/CoreInstallerBundle/Command/UpgradeCommand.php:108
 Zikula\Bundle\CoreInstallerBundle\Command\UpgradeCommand->execute() at /home/Kaik/public_html/mojeleicester/src/vendor/symfony/symfony/src/Symfony/Component/Console/Command/Command.php:253
 Symfony\Component\Console\Command\Command->run() at /home/Kaik/public_html/mojeleicester/src/vendor/symfony/symfony/src/Symfony/Component/Console/Application.php:882
 Symfony\Component\Console\Application->doRunCommand() at /home/Kaik/public_html/mojeleicester/src/vendor/symfony/symfony/src/Symfony/Component/Console/Application.php:195
 Symfony\Component\Console\Application->doRun() at /home/Kaik/public_html/mojeleicester/src/vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Console/Application.php:96
 Symfony\Bundle\FrameworkBundle\Console\Application->doRun() at /home/Kaik/public_html/mojeleicester/src/vendor/symfony/symfony/src/Symfony/Component/Console/Application.php:126
 Symfony\Component\Console\Application->run() at /home/Kaik/public_html/mojeleicester/src/app/console:19


zikula:upgrade [--username="..."] [--password="..."] [--router.request_context.host="..."] [--router.request_context.scheme="..."] [--router.request_context.base_url="..."]
```
I was able to trace it to https://github.com/zikula/core/blob/1.4/src/lib/Zikula/Bundle/CoreInstallerBundle/Controller/AjaxUpgradeController.php#L115

Looks like there is no admin category with offset 0 I have checked why this did not act on "System" category and in my db it was set to "1.System" I have changed that to "System" before another shoot and... 

```
[Kaik@kaikbig src]$ php app/console zikula:upgrade -vvv
---------------------------
| Zikula Upgrader Script |
---------------------------
*** UPGRADING TO ZIKULA CORE v1.4.0 ***
Please enter the username of the new Zikula admin user default:[admin]: Kaik
Please enter the password of the new Zikula admin user: 
Please enter the host where you install Zikula, e.g. "example.com". Do not include subdirectories: ml.dev.com
Please enter the scheme of where you install Zikula, can be either "http" or "https" default:[http]: 
Please enter the url path of the directory where you install Zikula, leave empty if you install it at the top level. Example: /my/sub-dir default:[]: 
Login
Logged in as admin
Upgrade modules
Modules upgraded
Install Zikula Routes Module
Zikula Routes Module installed
Reload routes
Routes reloaded
Regenerate themes
Themes regenerated
Finalize parameters
Parameters finalized
Clear caches
Caches cleared
Finish
Finish
UPGRADE COMPLETE!

```
So [0] is not working

**Comment by craigh** [comment]
please try again with the change I made.

**Comment by Kaik** [comment]
Now this showed up
```
[Kaik@kaikbig src]$ php app/console zikula:upgrade -vvv                                                                                
---------------------------
| Zikula Upgrader Script |
---------------------------
*** UPGRADING TO ZIKULA CORE v1.4.0 ***
Please enter the username of the new Zikula admin user default:[admin]: Kaik
Please enter the password of the new Zikula admin user: 
Please enter the host where you install Zikula, e.g. "example.com". Do not include subdirectories: ml.dev.com
Please enter the scheme of where you install Zikula, can be either "http" or "https" default:[http]: 
Please enter the url path of the directory where you install Zikula, leave empty if you install it at the top level. Example: /my/sub-dir default:[]: 
Login
Logged in as admin
Upgrade modules
Modules upgraded
Install Zikula Routes Module


                                    
  [InvalidArgumentException]        
  Invalid arguments array received  
                                    


Exception trace:
 () at /home/Kaik/public_html/mojeleicester/src/system/Zikula/Module/AdminModule/Api/AdminApi.php:253
 Zikula\Module\AdminModule\Api\AdminApi->addmodtocategory() at n/a:n/a
 call_user_func() at /home/Kaik/public_html/mojeleicester/src/lib/util/ModUtil.php:1175
 ModUtil::exec() at /home/Kaik/public_html/mojeleicester/src/lib/util/ModUtil.php:1251
 ModUtil::apiFunc() at /home/Kaik/public_html/mojeleicester/src/lib/Zikula/Bundle/CoreInstallerBundle/Controller/AjaxUpgradeController.php:116
 Zikula\Bundle\CoreInstallerBundle\Controller\AjaxUpgradeController->installRoutesModule() at /home/Kaik/public_html/mojeleicester/src/lib/Zikula/Bundle/CoreInstallerBundle/Controller/AjaxUpgradeController.php:70
 Zikula\Bundle\CoreInstallerBundle\Controller\AjaxUpgradeController->executeStage() at /home/Kaik/public_html/mojeleicester/src/lib/Zikula/Bundle/CoreInstallerBundle/Controller/AjaxUpgradeController.php:54
 Zikula\Bundle\CoreInstallerBundle\Controller\AjaxUpgradeController->commandLineAction() at /home/Kaik/public_html/mojeleicester/src/lib/Zikula/Bundle/CoreInstallerBundle/Command/UpgradeCommand.php:108
 Zikula\Bundle\CoreInstallerBundle\Command\UpgradeCommand->execute() at /home/Kaik/public_html/mojeleicester/src/vendor/symfony/symfony/src/Symfony/Component/Console/Command/Command.php:257
 Symfony\Component\Console\Command\Command->run() at /home/Kaik/public_html/mojeleicester/src/vendor/symfony/symfony/src/Symfony/Component/Console/Application.php:882
 Symfony\Component\Console\Application->doRunCommand() at /home/Kaik/public_html/mojeleicester/src/vendor/symfony/symfony/src/Symfony/Component/Console/Application.php:195
 Symfony\Component\Console\Application->doRun() at /home/Kaik/public_html/mojeleicester/src/vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Console/Application.php:96
 Symfony\Bundle\FrameworkBundle\Console\Application->doRun() at /home/Kaik/public_html/mojeleicester/src/vendor/symfony/symfony/src/Symfony/Component/Console/Application.php:126
 Symfony\Component\Console\Application->run() at /home/Kaik/public_html/mojeleicester/src/app/console:19


zikula:upgrade [--username="..."] [--password="..."] [--router.request_context.host="..."] [--router.request_context.scheme="..."] [--router.request_context.base_url="..."]



```
 Var is present and set to 5 and category with this ID exist. Can this be due to wrong var type? https://github.com/zikula/core/blob/1.4/src/system/Zikula/Module/AdminModule/Api/AdminApi.php#L252

**Comment by Guite** [comment]
`is_numeric()` returns `true` not only for numbers, but also for numeric strings.

**Comment by Kaik** [comment]
@Guite Hmm this is wired I thought same as you said but just checked by adding (int) typecasting and it is working 
   \ModUtil::apiFunc('ZikulaAdminModule', 'admin', 'addmodtocategory', array('module' => $routeModuleName, 'category' => (int)$destinationCategoryId));

**Comment by Guite** [comment]
Okay, weird indeed. I am going to commit this change.

**Comment by Guite** [comment]
Fixed in 1cd7dd7cf76ddb06f5775777778aa3fe1ab3b4b9

@Kaik thanks for testing!

**Comment by craigh** [comment]
:+1:
